src/de/darmstadt/tu/crossing/DigestInputStream.cryptsl

```diff
@@ -0,0 +1,33 @@
+SPEC java.security.DigestInputStream
+OBJECTS
+    boolean on;
+    byte[] b;
+    int off;
+    int len;
+    java.io.InputStream is;
+    java.security.MessageDigest md;
+
+EVENTS
+    c1: DigestInputStream(is, md);
+
+    g: md = getMessageDigest();
+
+    o: on(on);
+
+    r1: read();
+    r2: read(b, off, len);
+    Reads := r1 | r2;
+
+    c: close();
+
+FORBIDDEN
+    o;
```

Comment:
That's not how to specify forbidden methods. Please check PBEKeySpec for reference.

Reply:
"I declared the method in the EVENTS"
Yup, that is the problem. When you look at how the two PBEKeySpec constructors are declared, you'll notice their declarations are in the FORBIDDEN block directly. Their parameters are also not mapped to an object in the OBJECTS section above, but instead only type declaration. Please adopt that structure here, too.

Reply:
Ah ok, got it. Thanks very much

Reply:
Please push what you have to your branch. I'll check it out then.

Reply:
Ok, sure